arduino/libraries/librariesmanager/install_test.go

```diff
@@ -62,3 +63,43 @@ func TestParseGitURL(t *testing.T) {
 	require.Equal(t, "arduino-lib", libraryName)
 	require.NoError(t, err)
 }
+
+func TestValidateLibrary(t *testing.T) {
+	tmpDir := paths.New(t.TempDir())
+
+	nonExistingDirLib := tmpDir.Join("nonExistingDirLib")
+	err := validateLibrary(nonExistingDirLib)
+	require.Errorf(t, err, "directory doesn't exist: %s", nonExistingDirLib)
+
+	emptyLib := tmpDir.Join("emptyLib")
+	emptyLib.Mkdir()
+	err = validateLibrary(emptyLib)
+	require.Errorf(t, err, "library not valid")
+
+	onlyPropertiesLib := tmpDir.Join("onlyPropertiesLib")
+	onlyPropertiesLib.Mkdir()
+	onlyPropertiesLib.Join("library.properties").WriteFile([]byte{})
+	err = validateLibrary(onlyPropertiesLib)
+	require.Errorf(t, err, "library not valid")
+
+	onlyHeaderLib := tmpDir.Join("onlyHeaderLib")
```

Comment:
```suggestion
	onlyHeaderLib := tmpDir.Join("missingPropertiesLib")
```
I think this is a better description of what the library is intended to demonstrate (since an "only header" library can be valid if it has flat layout).